ensembldl.sh

```diff
@@ -0,0 +1,300 @@
+verbose=no
+
+for arg in "$@"
+do
+  case "$arg" in
+    -\?|--help)
+      usage
+      exit
+      ;;
+
+    --version)
+      echo "$app $version"
+      exit
+      ;;
+
+    -p|--parallel)
+      shift
+      cores=${1:?"parallel option has no argument"}
+      [[ $cores =~ ^[0-9]+$ ]] ||
+        bailout "parallel option argument is not a number: $cores"
+      shift
+      ;;
+
+    -q|--quiet)
+      verbose=no
+      shift
+      ;;
+
+    -v|--verbose)
+      verbose=yes
+      shift
+      ;;
+
+    --)
+      shift
+      break
+      ;;
+
+    -*)
+      bailout "unrecognized option: $1"
+      ;;
+
+    *)
+      break
+      ;;
+  esac
+done
+
+set +o nounset
+prefix=$1
+shift || bailout "missing argument: prefix"
+set -o nounset
+
+shopt -s extglob
+# trim trailing slashes
+prefix="${prefix%%+(/)}"
+
+datasets=()
+for d in "$@"
+do
+  datasets+=("${d%%+(/)}")
+done
+
+shopt -u extglob
+
+[[ ${#datasets[@]} -gt 0 ]] ||
+  bailout 'missing argument: dataset'
+
+# -----------------------------------------------------------------------------
+# verbose: output configuration
+# -----------------------------------------------------------------------------
+
+if [[ $verbose == yes ]]
+then
+  cat << EOF
+prefix: $prefix
+datasets:
+EOF
+  for d in "${datasets[@]}"
+  do
+    echo "- $d"
+  done
+  cat << EOF
+parallel: $cores CPU cores
+
+versions:
+- $app $version
+- $(lftp --version | head -1)
+
+EOF
+fi
+
+# -----------------------------------------------------------------------------
+# check arguments
+# -----------------------------------------------------------------------------
+
+[[ -d $prefix ]] ||
+  bailout "local data set directory does not exist: $prefix"
+
+download_date=$(date +%F)
+
+# -----------------------------------------------------------------------------
+# preparation
+# -----------------------------------------------------------------------------
+
+tmpdir=$(mktemp -d --tmpdir="$prefix" ".$app-XXXXXXXXXX")
+trap 'rm -fr "$tmpdir"' EXIT INT TERM
+
+# -----------------------------------------------------------------------------
+# application functions
+# -----------------------------------------------------------------------------
+
+function download {
+  local v
+  [[ $verbose == yes ]] &&
+    v=y
+
+  cat << EOF | lftp ftp://ftp.ensembl.org
+# bigger socket buffer, better I/O
+set net:socket-buffer 33554432
+
+# use IPv4 only
+set dns:order "inet"
+
+# download md5s first
+mirror ${v:+-v} -r -p -P $cores -i \
+       "^(CHECKSUMS|MD5SUM)$" \
+       /$(dirname "$dataset") $tmpdir
+
+# then download tarballs
+mirror ${v:+-v} -r -p -P $cores -i \
+       "^$(basename "$dataset").*\\.gz$" \
+       /$(dirname "$dataset") $tmpdir
+EOF
+}
+
+# -----------------------------------------------------------------------------
+# application
+# -----------------------------------------------------------------------------
+
+for dataset in "${datasets[@]}"
+do
+  output_basedir="$prefix/ncbi/$dataset"
```

Comment:
This should also be changed to **ensembl** instead of **ncbi**, right?

Reply:
done